diff --git a/inc/i2c_ccs811sensor.h b/inc/i2c_ccs811sensor.h
index 7fec0fe..7fe71cb 100644
--- a/inc/i2c_ccs811sensor.h
+++ b/inc/i2c_ccs811sensor.h
@@ -6,11 +6,9 @@
 
 #include "FreeRTOS.h"
 
-struct co2_tvoc {
-	double co2, tvoc;
-};
+double get_tvoc(void);
+double get_co2(void);
 
-struct co2_tvoc get_co2_tvoc(void);
 void i2c_ccs811sensor(void *pvParameters);
 
 #endif 
diff --git a/src/data_collector.c b/src/data_collector.c
index 8b44803..9316ebb 100644
--- a/src/data_collector.c
+++ b/src/data_collector.c
@@ -5,6 +5,7 @@
 #include "data_structure.h"
 #include "eth_sender.h"
 #include "lm335z.h"
+#include "i2c_ccs811sensor.h"
 #include "uart_sensors.h"
 #include "fans.h"
 
@@ -48,6 +49,10 @@ void data_collector(void *pvParameters)
         dataPacets_buffer[current_packet].hcho   = get_HCHO(); 
         dataPacets_buffer[current_packet].pm10   = get_pm10();
         dataPacets_buffer[current_packet].pm2_5  = get_pm2_5();
+
+        dataPacets_buffer[current_packet].co2   = get_co2();
+        dataPacets_buffer[current_packet].tvoc  = get_tvoc();
+
         //dataPacets_buffer[current_packet].humidity=;
         //dataPacets_buffer[current_packet].pressure=;
         //dataPacets_buffer[current_packet].tvoc=;
@@ -71,7 +76,8 @@ void avrg_data_packets(dataPacket_S *buffer, int buf_size, dataPacket_S *result_
         result_packet->pm2_5    += (buffer[i].pm2_5); 
         // result_packet->humidity += (buffer[i].humidity); 
         // result_packet->pressure += (buffer[i].pressure); 
-        // result_packet->tvoc     += (buffer[i].tvoc); 
+         result_packet->tvoc     += (buffer[i].tvoc); 
+         result_packet->co2     += (buffer[i].co2); 
     }
         result_packet->co != 0 ? result_packet->co /= buf_size : 0;
         result_packet->so2 != 0 ? result_packet->so2 /= buf_size : 0;
@@ -83,5 +89,6 @@ void avrg_data_packets(dataPacket_S *buffer, int buf_size, dataPacket_S *result_
         result_packet->pm2_5 != 0 ? result_packet->pm2_5 /= buf_size : 0;
         // result_packet->humidity /= buf_size; 
         // result_packet->pressure /= buf_size; 
-        // result_packet->tvoc     /= buf_size; 
+        result_packet->tvoc     /= buf_size; 
+        result_packet->co2     /= buf_size; 
 }
diff --git a/src/i2c_ccs811sensor.c b/src/i2c_ccs811sensor.c
index e72613c..937151b 100644
--- a/src/i2c_ccs811sensor.c
+++ b/src/i2c_ccs811sensor.c
@@ -6,7 +6,8 @@
 #include "task.h"
 #include "wh1602.h"
 
-struct co2_tvoc co2_tvoc_t;
+double co2;
+double tvoc2;
 
 I2C_HandleTypeDef  hi2cxc;
 
@@ -73,8 +74,8 @@ void readResults()
 
 	/*	TVOC value, in parts per billion (ppb)
 		eC02 value, in parts per million (ppm) */
-	co2_tvoc_t.co2 = ((unsigned int)co2MSB << 8) | co2LSB;
-	co2_tvoc_t.tvoc = ((unsigned int)tvocMSB << 8) | tvocLSB;
+	co2 = ((unsigned int)co2MSB << 8) | co2LSB;
+	tvoc = ((unsigned int)tvocMSB << 8) | tvocLSB;
 }
 
 /*
@@ -234,9 +235,14 @@ void writeRegister(uint8_t addr, uint8_t val)
 }
 
 
-struct co2_tvoc get_co2_tvoc(void)
+double get_tvoc(void)
 {
-	return co2_tvoc_t;
+	return tvoc;
+}
+
+double get_co2_tvoc(void)
+{
+	return co2;
 }
 
 void i2c_ccs811sensor(void *pvParameters) 
